src/plasmapy/simulation/particle_tracker/particle_tracker.py

```diff
@@ -620,117 +703,205 @@ def particles_on_grid(self):
 
         return all_particles
 
-    def _push(self) -> None:
+    @property
+    def _particle_kinetic_energy(self):
         r"""
-        Advance particles using an implementation of the time-centered
-        Boris algorithm.
+        Return the non-relativistic kinetic energy of the particles.
         """
+
+        # TODO: how should the relativistic case be handled?
+        return 0.5 * self.m * np.square(np.linalg.norm(self.v, axis=-1, keepdims=True))
+
+    def _interpolate_grid(self):
         # Get a list of positions (input for interpolator)
         tracked_mask = self._tracked_particle_mask
 
         self.iteration_number += 1
 
-        pos_all = self.x
-        pos_tracked = pos_all[tracked_mask]
-
-        vel_all = self.v
-        vel_tracked = vel_all[tracked_mask]
+        pos_tracked = self.x[tracked_mask]
 
         # entered_grid is zero at the end if a particle has never
         # entered any grid
         self.entered_grid += np.sum(self.particles_on_grid, axis=-1).astype(np.bool_)
 
-        Ex = np.zeros(self.nparticles_tracked) * u.V / u.m
-        Ey = np.zeros(self.nparticles_tracked) * u.V / u.m
-        Ez = np.zeros(self.nparticles_tracked) * u.V / u.m
-        Bx = np.zeros(self.nparticles_tracked) * u.T
-        By = np.zeros(self.nparticles_tracked) * u.T
-        Bz = np.zeros(self.nparticles_tracked) * u.T
+        # TODO: how should we handle unrecognized quantities?
+
+        # Construct a dictionary to store the interpolation results
+        # This is accomplished using a dictionary comprehension. Each quantity is
+        # initialized as a zeros array with its respective units
+        summed_field_values = {
```

Comment:
Name confused me for a bit - this is the total field across all grids? Maybe total_grid_values? 

Also field->grid or something, since it's not just the fields anymore (rho)